README.md

```diff
@@ -4,6 +4,36 @@
 A [Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
 implementation for the [Smithy IDL](https://awslabs.github.io/smithy/).
 
+
+### Running the LSP
+
+There are three ways to launch the LSP, and which you choose depends on your use case.
+
+In all cases, the communication protocol is JSON-RPC, the transport channels can are:
+
+#### Stdio
+
+Run `./gradlew run --args="0"`
+
+The LSP will use stdio (stdin, stdout) to communicate.
+
+#### Sockets
+
+Run `./gradlew run --args="12423"`
+
+The LSP will try to connect to the given port using a TCP socket - if it can't, it will fail.
+
+This is used by the VSCode extension to establish a connection between it and the LSP (which is launched
+as a local process)
+
+#### WebSockets
+
+Run `./gradlew run --args="3000 --ws"`
```

Comment:
Can you put `--ws` in front? I would assume here that `3000` is a positional argument that comes after named arguments, and those are usually last.

Reply:
perhaps it's also worth considering the deprecation of `port` as a positional argument and making it a `--port` option? For some time the server could support both styles, but eventually the positional way would be dropped.